docs: Add `clients` subdirective to `tls` directive docs (#49)
caddyserver/website: src/docs/markdown/caddyfile/directives/tls.md

	ca_root   <pem_file>
	dns       <provider_name> [<params...>]
	on_demand
	clients {
		mode [request|require|verify_if_given|require_and_verify]
		trusted_ca_certs        <trusted_ca_certs...>
		trusted_ca_certs_file   <filename...>
		trusted_leaf_certs      <trusted_leaf_certs...>
		trusted_leaf_certs_file <filename...>

> 	}

Reviewer: The syntax has changed a little bit recently: https://github.com/caddyserver/caddy/pull/3335/files#diff-3235e86b75f6440c822f6d7d8f4beb61R65-R71

> 	clients {
> 		mode [request|require|verify_if_given|require_and_verify]
> 		trusted_ca_certs        <trusted_ca_certs...>
> 		trusted_ca_certs_file   <filename...>
> 		trusted_leaf_certs      <trusted_leaf_certs...>
> 		trusted_leaf_certs_file <filename...>
> 	}

Reviewer: ```suggestion
	client_auth {
		mode                   [request|require|verify_if_given|require_and_verify]
		trusted_ca_cert        <base64_der>
		trusted_ca_cert_file   <filename>
		trusted_leaf_cert      <base64_der>
		trusted_leaf_cert_file <filename>
	}
```

- **ca_root** specifies a PEM file that contains a trusted root certificate for the ACME CA endpoint, if not in the system trust store.
- **dns** enables the [DNS challenge](/docs/automatic-https#dns-challenge) using the specified provider plugin, which must be plugged in from one of the [caddy-dns](https://github.com/caddy-dns) repositories. Each provider plugin may have their own syntax following their name; refer to their docs for details. Maintaining support for each DNS provider is a community effort. [Learn how to enable the DNS challenge for your provider at our wiki.](https://caddy.community/t/how-to-use-dns-provider-modules-in-caddy-2/8148)
- **on_demand** enables [on-demand TLS](/docs/automatic-https#on-demand-tls) for the hostnames given in the site block's address(es).
- **clients** enables and configures TLS client authentication.

The `clients` block can look like this:

```caddy-d
clients {
	mode                    [request|require|verify_if_given|require_and_verify]
	trusted_ca_certs        <trusted_ca_certs...>
	trusted_ca_certs_file   <filename...>
	trusted_leaf_certs      <trusted_leaf_certs...>
	trusted_leaf_certs_file <filename...>
}

> ```

Reviewer: This section will need updating as well.

- **trusted_ca_certs** is a list of base64 DER-encoded CA certificates against which to validate client certificates. Client certs which are not signed by any of these CAs will be rejected.
- **trusted_ca_certs_file** is a list of base64 DER-encoded CA certificate files against which to validate client certificates. Client certs which are not signed by any of these CAs will be rejected.
- **trusted_leaf_certs** is a list of base64 DER-encoded client leaf certs to accept. If this list is not empty, client certs which are not in this list will be rejected

> - **trusted_leaf_certs_file** is a list of base64 DER-encoded CA certificate files against which to validate client certificates. Client certs which are not signed by any of these CAs will be rejected.

Reviewer: This needs updating too

- **mode** is the mode for authenticating the client. Allowed values are:
  | Mode               | Description                                                                              |
  |--------------------|------------------------------------------------------------------------------------------|
  | request            | Ask clients for a certificate, but allow even if there isn't one; do not verify it       |
  | require            | Require clients to present a certificate, but do not verify it                           |
  | verify_if_given    | Ask clients for a certificate; allow even if there isn't one, but verify it if there is  |
  | require_and_verify | Require clients to present a valid certificate that is verified                          |

> 	The default mode is `require_and_verify` if any TrustedCACerts or TrustedLeafCerts are provided; otherwise, the default mode is `require`

Reviewer: ```suggestion
	The default mode is `require_and_verify` if any `trusted_ca_cert` or `trusted_leaf_cert` are provided; otherwise, the default mode is `require`.
```

tls {
	dns cloudflare {env.CLOUDFLARE_API_TOKEN}
}
```

> Enable TLS Client Authentication

Reviewer: Explain what kind of authentication, or what is necessary and sufficient to auth

```caddy-d
tls {
	clients {
		mode                    require_and_verify

> 		trusted_ca_certs_file   cacerts.crt

Reviewer: ```suggestion
		trusted_ca_cert_file   cacerts.crt
```

> tls {
>   clients {
>     mode                    require_and_verify
>     trusted_ca_certs_file   cacerts.crt
>   }
> }

Reviewer 2: Spaces -> tabs

```suggestion
tls {
	clients {
		mode                    require_and_verify
		trusted_ca_certs_file   cacerts.crt
	}
}
```
